### DISCUSSION TITLE
Create a timeline element  with Zeitline

### DISCUSSION THREAD

**Discussion by dridk** [zauberzeug/nicegui#3620] [Ideas / Feature Requests]
It would be helpful to create a timeline element using zeitline library. 
https://octree-gva.github.io/Zeitline/docs/examples/base.html

```python
ui.zeitline(date_range=(..),  data = [ date1, date2 ... ] )


```

**Comment by rodja** [comment]
Looks cool. Anyone interested in creating a pull request?

**Comment by Aluriak** [comment]
Yeah, i would be. I am however not js dev, so i need some pointer.

It seems that zeitline is based on react. Is it possible to integrate it to nicegui nonetheless ?

I will show my attempt in a PR.

**Reply by dridk** [reply]
I don't think so. It is build on top of d3.js.  The project is shipped with a React component.  
 https://unpkg.com/zeitline@0.1.0/dist/zeitline.bundle.min.js.

**Reply by dridk** [reply]
Check echart component.  I guess you have to create zeitline.js ( create a vue.js component) and zeitline.py. 

https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/echart.py
https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/echart.js
